param/parameterized.py

```diff
@@ -258,33 +258,70 @@ def _f(self, obj, val):
     return _f
 
 
+def ismethod(obj):
+    if sys.version_info.major == 2:
+        return hasattr(obj, 'im_class')
+    else:
+        owner = getattr(inspect.getmodule(meth),
+                        meth.__qualname__.split('.<locals>', 1)[0].rsplit('.', 1)[0])
+        return isinstance(owner, type)
```

Comment:
Have to think about this, since py3 does not have the concept of an unbound method we need this hack to find out whether we are decorating a simple function or not. It works well but I'm not entirely convinced it's perfectly robust. If we give up on this then two things would need to happen:

1. We either do not allow keywords at all (only args) or we pointlessly also allow them for methods
2. We do not support ``watch=True`` for functions.